src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.LogicApps.tests.ps1

```diff
@@ -678,4 +710,462 @@ InModuleScope Arcus.Scripting.LogicApps {
+            It "Doesn't disable anything when checkType = NoWaitingOrRunningRuns but returns a running run for unknown stopType" {
+                # Arrange
+                $script:i = 0
+                $resourceGroup = "my-resource-group"
+                $logicAppNames = @("snd-async", "ord-sthp-harvest-order-doublechecker", "ord-sthp-harvest-order-doublechecker")
+
+                Mock Invoke-WebRequest -MockWith {
+                    $script:i++
+                    if ($script:i -gt 2) {
+                        Write-Host "Returning empty (no running, no waiting) runs"
+                        return [pscustomobject] @{
+                            Content = '{"value":[]}'
+                        }
+                    } else {
+                        Write-Host "Returning 1 running run"
+                        return [pscustomobject] @{
+                            Content = '{"value":[{"properties":{"status":"Running"}}]}'
+                        }
+                    }
+                }
+                Mock Disable-AzLogicApp {}
+                Mock Get-AzCachedAccessToken -MockWith {
+                    $Global:subscriptionId = '123456'
+                    return @{
+                        SubscriptionId = "123456"
+                        AccessToken = "accessToken"
+                    }
+                }
+
+                # Act
+                Disable-AzLogicAppsFromConfig -DeployFileName "$PSScriptRoot\Files\LogicApps\Standard\deploy-orderControl-noWaitingOrRunningRunsWithunknownStopType.json" -ResourceGroupName $resourceGroup
+
+                # Assert
+                Assert-MockCalled Get-AzCachedAccessToken -Scope It -Times 1
+                Assert-MockCalled Invoke-WebRequest -Scope It -Exactly 8
+                Assert-MockCalled Disable-AzLogicApp -Scope It -Exactly 0
+            }
+            It "Doesn't disable anything when checkType = NoWaitingOrRunningRuns but returns a waiting run for stopType = None" {
+                # Arrange
+                $script:i = 0
+                $resourceGroup = "my-resource-group"
+                $logicAppNames = @("snd-async", "ord-sthp-harvest-order-doublechecker")
+
+                Mock Invoke-WebRequest -MockWith {
+                    $script:i++
+                    if ($script:i -gt 2) {
+                        Write-Host "Returning empty (no running, no waiting) runs"
+                        return [pscustomobject] @{
+                            Content = '{"value":[]}'
+                        }
+                    } else {
+                        Write-Host "Returning 1 waiting run"
+                        return [pscustomobject] @{
+                            Content = '{"value":[{"properties":{"status":"Waiting"}}]}'
+                        }
+                    }
+                }
+                Mock Disable-AzLogicApp {}
+                Mock Get-AzCachedAccessToken -MockWith {
+                    $Global:subscriptionId = '123456'
```

Comment:
This can now be removed?

Reply:
Not yet, needed because this script uses the global variables. We will keep this in for now and in the future remove it 👍🏻